### PR TITLE
feat(auth-middleware): don't auth OPTIONS and HEAD requests for SessionAuth middlerware

### LINKED ISSUE #2181: Bug: regression v2.0.0beta4 -> main@HEAD
### Description

There appears to be a regression between tag v2.0.0beta4 and the current (2.0.0rc1 PyPi release) (or if you prefer, also commit d257d099993179f1133a96441bf816e1de435007, which is the latest commit on main. This seems very likely related to #2160  which my guess would be only fixed the problem for JWT, not the SessionAuth middleware that I am using.

### URL to code causing the issue

https://github.com/Mattwmaster58/MacTrack/tree/broken-auth

### MCVE

```python
(working on this)
```

### Steps to reproduce

```bash
1. Clone repo, checkout `broken-auth` branch, poetry install, run, request to OPTIONS endpoint, observe:

`INFO:     127.0.0.1:55754 - "OPTIONS /api/filters/create HTTP/1.1" 401 Unauthorized`
```

### Screenshots

_No response_

### Logs

```bash
ERROR - 2023-08-19 18:27:03,266908605 - litestar - middleware - exception raised on http connection to route /api/filters/create

Traceback (most recent call last):
  File "/home/user/Documents/projects/mac-track/api/.venv/lib/python3.11/site-packages/litestar/middleware/exceptions/middleware.py", line 191, in __call__
    await self.app(scope, receive, send)
  File "/home/user/Documents/projects/mac-track/api/.venv/lib/python3.11/site-packages/litestar/middleware/authentication.py", line 87, in __call__
    auth_result = await self.authenticate_request(ASGIConnection(scope))
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/Documents/projects/mac-track/api/.venv/lib/python3.11/site-packages/litestar/security/session_auth/middleware.py", line 111, in authenticate_request
    raise NotAuthorizedException("no session data found")
litestar.exceptions.http_exceptions.NotAuthorizedException: 401: no session data found
Traceback (most recent call last):
  File "/home/user/Documents/projects/mac-track/api/.venv/lib/python3.11/site-packages/litestar/middleware/exceptions/middleware.py", line 191, in __call__
    await self.app(scope, receive, send)
  File "/home/user/Documents/projects/mac-track/api/.venv/lib/python3.11/site-packages/litestar/middleware/authentication.py", line 87, in __call__
    auth_result = await self.authenticate_request(ASGIConnection(scope))
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/Documents/projects/mac-track/api/.venv/lib/python3.11/site-packages/litestar/security/session_auth/middleware.py", line 111, in authenticate_request
    raise NotAuthorizedException("no session data found")
litestar.exceptions.http_exceptions.NotAuthorizedException: 401: no session data found
INFO:     127.0.0.1:55754 - "OPTIONS /api/filters/create HTTP/1.1" 401 Unauthorized
```

### Litestar Version

2.0.0rc1 as well as d257d099993179f1133a96441bf816e1de435007

### Platform

- [X] Linux
- [ ] Mac
- [ ] Windows
- [ ] Other (Please specify in the description above)

---

## Funding
* If you would like to see an issue prioritized, make a pledge towards it!
* We receive the pledge once the issue is completed & verified

<a href="https://polar.sh/litestar-org/litestar/issues/2181">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/2181/pledge.svg?darkmode=1">
  Fund with Polar
</picture>
</a>

### PR DESCRIPTION
**Pull request by Mattwmaster58**
this breaks browser preflights on authenticated endpoints

### Pull Request Checklist

[x]: # "By submitting this issue, you agree to:"
[x]: # "- follow Litestar's [Code of Conduct](https://github.com/litestar-org/.github/blob/main/CODE_OF_CONDUCT.md)"
[x]: # "- follow Litestar's [Contribution Guidelines](https://litestar.dev/community/contribution-guide)"

- [x] New code has 100% test coverage
- [x] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [x] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR

### Description
Follow up on #2160 which didn't address SessionAuth middleware. 

### Close Issue(s)
[//]: # "Please add in issue numbers this pull request will close, if applicable"
[//]: # "Examples: Fixes #4321 or Closes #1234"

- Fixes #2181

### CODE REVIEW

**Comment by Mattwmaster58** [comment]
there were some mypy issues, but most seemed unrelated to my code

**Review by cofin — APPROVED** [review]
LGTM.  I'll see if we can get these other errors cleaned up.

File: tests/unit/test_middleware/test_session/test_integration.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    password: SecretStr
+
+
+MOCK_DB: dict[str, User] = {}
+memory_store = MemoryStore()
+
+
+async def retrieve_user_handler(
+    session: dict[str, Any], connection: "ASGIConnection[Any, Any, Any, Any]"
+) -> User | None:
```
Comment (provinzkraut) [inline tests/unit/test_middleware/test_session/test_integration.py:36]:
```suggestion
) -> Optional[User]:
```

File: tests/unit/test_middleware/test_session/test_integration.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
+from typing import Any, Literal
```
Comment (provinzkraut) [inline tests/unit/test_middleware/test_session/test_integration.py:1]:
```suggestion
from typing import Any, Literal, Optional
```

**Review by provinzkraut — CHANGES_REQUESTED** [review]

**Pull request merged**